etc/WMAgentConfig.py

```diff
@@ -356,3 +356,18 @@
 config.AgentStatusWatcher.userAMQ = "OVERWRITE_BY_SECRETS"
 config.AgentStatusWatcher.passAMQ = "OVERWRITE_BY_SECRETS"
 config.AgentStatusWatcher.topicAMQ = "OVERWRITE_BY_SECRETS"
+
+config.component_("RucioInjector")
+config.RucioInjector.namespace = "WMComponent.RucioInjector.RucioInjector"
+config.RucioInjector.componentDir = config.General.workDir + "/RucioInjector"
+config.RucioInjector.logLevel = globalLogLevel
+config.RucioInjector.enabled = False
+config.RucioInjector.pollInterval = 300
+config.RucioInjector.pollIntervalRules = 43200
+config.RucioInjector.cacheExpiration = 24 * 60 * 60  # one day
+config.RucioInjector.createBlockRules = True
+config.RucioInjector.useTestRSE = ""  # set it to "_Test" in a test node
+config.RucioInjector.skipRulesForTiers = ["NANOAOD", "NANOAODSIM"]
```

Comment:
This somehow seems backwards to me, maybe I am missing something. I expect that we *would* be making rules for injected NanoAOD and *not* for any other data tiers as those are expected to be handled by PhEDEx for now, where the sync daemon would keep phedex and rucio in agreement about where it is. (we can still inject replicas for these I think, as the sync script will just see the work it was going to do done already and not do any harm, i.e. there's a race between WMCore and sync but either one winning produces the same state)

So I would think we want something like `makeRulesForTiers = ["NANOAOD", "NANOAODSIM"]` and the list grows as we progress until it is short-circuited at some point to always true once we no longer inject anything into phedex.

Reply:
Now, aside from all that, there is the question whether WMCore needs to make a rule at all. In theory, no since a rucio subscription can do that. However I think that ATLAS puts a rule with something like 1w lifetime on output so we might want to keep that option open.

Reply:
Nick, what I had understood from our discussion was:
* WMCore will inject only NANOAOD/NANOAODSIM data into Rucio (container + block + replicas)
* and rules - for those 2 datatiers - would be sorted by the DM team, given that we won't know how much of our container (cms dataset) is/was injected into PhEDEx and how much into Rucio.

BTW, the ban on data injection is not yet in place. I have it implemented only for rules for now. I actually want to work on that front in a different PR, see this issue:
https://github.com/dmwm/WMCore/issues/9600

Reply:
I don't think there is any need for a ban on data injection, its a race between the sync script that always end up with the same result. @ericvaandering thoughts?

Reply:
There will be a moment that we will keep running PhEDExInjector and RucioInjector. Whatever one component does, the other cannot. Reason is because we keep the state of blocks/files in the local database and we cannot (should not) have two different components trying to do the same database updates.
That's why, IMO, whatever we decide to inject into Rucio, we need to NOT inject to PhEDEX - at least from WMAgent point of view.

Reply:
Ok that is reasonable.
For the rules, do I understand that its possible during the transition that part of a CMS dataset will be injected in PhEDEx and part in Rucio? Will we still create the Rucio container (CMS dataset) rule?

Reply:
Yes, it's possible. We have workflows that remain in running-* state in the system for many months. So it could easily have blocks created/injected against PhEDEx, while others are in a new agent version injecting things into Rucio.
Here indeed we would have a race between sync and rucio injected data. However, RucioInjector handles nicely existent containers and just move on with block/replicas injection.